scripts/test_stale_sources.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Test if there are any stale java source files
+
+# Compile first once to ensure we have compiled everything
+mvn_output=$(mvn -X test-compile)
+
+# Compile again to find stale sources
+mvn_output=$(mvn -X test-compile)
```

Comment:
Just out of curiosity, what causes the need to compile twice?

Reply:
If you do a `mvn clean`, then all the java files are stale. So it's on the second compile you can tell if there are stale sources.

This step is probably not needed since the compile is done in the static analysis CI before this script is called. But if someone later changes things, this steps ensures that the sources are compiled twice. It's a case of belt and braces to ensure everything works in case the unexpected happens.